cve_bin_tool/checkers/rsync.py

```diff
@@ -0,0 +1,52 @@
+"""
+CVE checker for rsync
+References:
+https://www.cvedetails.com/vulnerability-list/vendor_id-7059/Rsync.html
+"""
+import re
+
+def guess_version(lines):
+    """ Guesses the rsync version from the file contents """
+    new_guess = ""
+    pattern1 = re.compile(r"[0-3]+\.[0-9]+\.[0-9]+")
+
+    for line in lines:
+        match = pattern1.search(line)
+        if match:
+            new_guess2 = match.group().strip()
+            if len(new_guess2) > len(new_guess):
+                new_guess = new_guess2
+    return new_guess
+
+def guess_contains(lines):
+    """
+    Tries to determine if a file includes rsync.
+    """
+    signatures = [
+        r"Invalid checksum count %ld [%s]",
+        r"Hlink node data for %d is NULL (%s)",
+        r"decompressor lost sync!",
+    ]
+
+    for line in lines:
+        for signature in signatures:
+            pattern = re.compile(signature)
+            if pattern.search(line):
+                return 1
+    return 0
+
+def get_version(lines, filename):
+    """returns version information for rsync as found in a given file.
+    VPkg: rsync, rsync
```

Comment:
You probably want to detect the stuff for Samba Rsync as well as the stuff with Rsync marked as the vendor.  Read the references to be sure, but it looks to me like they refer to the same code base, and it's definitely the samba version that's installed by default on my system when I do a rsync --version.